executor/join.go

```diff
@@ -74,11 +79,26 @@ type HashJoinExec struct {
 
 	memTracker *memory.Tracker // track memory usage.
 
-	// radixBits indicates the bit number using for radix partitioning. Inner
-	// relation will be split to 2^radixBits sub-relations before building the
-	// hash tables. If the complete inner relation can be hold in L2Cache in
-	// which case radixBits will be 0, we can skip the partition phase.
-	radixBits int
+	// radixBits indicates the bits using for radix partitioning. Inner relation
+	// will be split to 2^radixBitsNumber sub-relations before building the hash
+	// tables. If the complete inner relation can be hold in L2Cache in which
+	// case radixBits will be 1, we can skip the partition phase.
+	radixBits  uint32
+	innerParts []partition
+	// innerRowPrts indicates the position in corresponding partition of every
+	// row in innerResult.
+	innerRowPrts    [][]ptr4Partition
+	partConcurrency int
+}
+
+// partition stores the sub-relations of inner relation and outer relation after
+// partition phase. Every partition can be fully stored in L2 cache.
+type partition = *chunk.List
```

Comment:
The L2 cache size is not very big. How about use `Chunk` to directly store all the data inside a partition continuously in the memory?